eessi/testsuite/tests/apps/lammps/lammps.py

```diff
@@ -0,0 +1,168 @@
+
+        return sn.assert_eq(n_atoms, 32000)
+
+    @run_after('init')
+    def run_after_init(self):
+        """hooks to run after init phase"""
+
+        # Filter on which scales are supported by the partitions defined in the ReFrame configuration
+        hooks.filter_supported_scales(self)
+
+        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.device_type)
+
+        hooks.set_modules(self)
+
+        # Set scales as tags
+        hooks.set_tag_scale(self)
+
+    @run_after('setup')
+    def run_after_setup(self):
+        """hooks to run after the setup phase"""
+        if self.device_type == 'cpu':
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT['CPU'])
+        elif self.device_type == 'gpu':
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT['GPU'])
+        else:
+            raise NotImplementedError(f'Failed to set number of tasks and cpus per task for device {self.device_type}')
+
+        # Set OMP_NUM_THREADS environment variable
+        hooks.set_omp_num_threads(self)
+
+
+@rfm.simple_test
+class EESSI_LAMMPS_lj(EESSI_LAMMPS_base):
+    tags = {TAGS['CI']}
+
+    sourcesdir = 'src/lj'
+    executable = 'lmp -in in.lj'
+
+    @deferrable
+    def check_number_neighbors(self):
+        '''Assert that the test calulated the right number of neighbours'''
+        regex = r'^Total # of neighbors = (?P<neigh>\S+)'
+        n_neigh = sn.extractsingle(regex, self.stdout, 'neigh', int)
+
+        return sn.assert_eq(n_neigh, 1202833)
+
+    @performance_function('timesteps/s')
+    def perf(self):
+        regex = r'^Performance: [.0-9]+ tau/day, (?P<perf>[.0-9]+) timesteps/s,'
+        return sn.extractsingle(regex, self.stdout, 'perf', float)
+
+    @sanity_function
+    def assert_sanity(self):
+        '''Check all sanity criteria'''
+        return sn.all([
+            self.assert_lammps_openmp_treads(),
+            self.assert_lammps_processor_grid(),
+            self.assert_run(),
+            self.check_number_neighbors(),
+        ])
+
+    @run_after('setup')
+    def set_executable_opts(self):
+        """Set executable opts based on device_type parameter"""
+        num_default = 0  # If this test already has executable opts, they must have come from the command line
+        hooks.check_custom_executable_opts(self, num_default=num_default)
+        if not self.has_custom_executable_opts:
+            # should also check if the lammps is installed with kokkos.
+            # Because this exutable opt is only for that case.
+            if self.device_type == "gpu":
+                self.executable_opts += [
+                    f'-kokkos on t {self.num_cpus_per_task} g {self.num_gpus_per_node}',
+                    '-suffix kk',
+                    '-package kokkos newton on neigh half',
+                ]
+                utils.log(f'executable_opts set to {self.executable_opts}')
```

Comment:
This works so than and than these options are only used for GPU build with kokkos and does a test that does not uses the GPUs. So now would also be to add the `executable_opts` for the CUDA enabled installation without kokkos.
```suggestion
                if 'kokkos' in self.module_name:
                    self.executable_opts += [
                        f'-kokkos on t {self.num_cpus_per_task} g {self.num_gpus_per_node}',
                        '-suffix kk',
                        '-package kokkos newton on neigh half',
                    ]
                    utils.log(f'executable_opts set to {self.executable_opts}')
```

Reply:
done